File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarFirmwareUpdateModels.swift
//  Copyright © 2024 Polar. All rights reserved.

import Foundation

public enum CheckFirmwareUpdateStatus {
    case checkFwUpdateAvailable(version: String)
    case checkFwUpdateNotAvailable(details: String)
    case checkFwUpdateFailed(details: String)
}

public enum FirmwareUpdateStatus {
    case fetchingFwUpdatePackage(details: String)
    case preparingDeviceForFwUpdate(details: String)
    case writingFwUpdatePackage(details: String)
    case finalizingFwUpdate(details: String)
    case fwUpdateCompletedSuccessfully(details: String)
    case fwUpdateNotAvailable(details: String)
    case fwUpdateFailed(details: String)
}

struct PolarFirmwareVersionInfo {
    let deviceFwVersion: String
    let deviceModelName: String
    let deviceHardwareCode: String
}

File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarFirmwareUpdateModels.kt
package com.polar.sdk.api.model

/**
 * Status for firmware update.
 *
 * @param details, extra details, if any
 */
sealed class FirmwareUpdateStatus(open val details: String = "") {
    data class FetchingFwUpdatePackage(override val details: String = "") : FirmwareUpdateStatus(details)
    data class PreparingDeviceForFwUpdate(override val details: String = "") : FirmwareUpdateStatus(details)
    data class WritingFwUpdatePackage(override val details: String = "") : FirmwareUpdateStatus(details)
    data class FinalizingFwUpdate(override val details: String = "") : FirmwareUpdateStatus(details)
    data class FwUpdateCompletedSuccessfully(override val details: String = "") : FirmwareUpdateStatus(details)
    data class FwUpdateNotAvailable(override val details: String = "") : FirmwareUpdateStatus(details)
    data class FwUpdateFailed(override val details: String = "") : FirmwareUpdateStatus(details)
}

data class PolarFirmwareVersionInfo(
        val deviceFwVersion: String,
        val deviceModelName: String,
        val deviceHardwareCode: String
)